libraries/standard/serializer/src/cbor/iot_serializer_tinycbor_decoder.c

```diff
@@ -467,6 +525,15 @@ static IotSerializerError_t _next( IotSerializerDecoderIterator_t iterator )
 
     cborError = cbor_value_advance( &pCborValueWrapper->cborValue );
 
+    if( cbor_value_at_end( &pCborValueWrapper->cborValue ) == false )
+    {
+        /* Update the iterator to store the encoded data pointer and length for the new element being pointed to. */
+        IotSerializerDecoderObject_t * pDecoderObject = ( IotSerializerDecoderObject_t * ) iterator;
+        pDecoderObject->bufferPtr = pCborValueWrapper->cborValue.ptr;
+        pDecoderObject->dataLengthInBuffer = calculateSizeOfCborObject(
```

Comment:
cbor_value_advance() is a costly operation for nested structures. Can we do it on demand from the API.

Reply:
Good suggestion. Based on our collective Chime discussion, we decided to remove support of obtaining buffer address and data size for iterator object (as decoder object should be sourced from iterator object through `_get()` operation)